examples/mqtt-paho/main.c

```diff
@@ -0,0 +1,231 @@
+ * @{
+ *
+ * @file
+ * @brief       MQTT-paho example
+ *
+ * @author      Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include "msg.h"
+#include "shell.h"
+#include "lwip/netif.h"
+#include "lwip.h"
+#include "mqtt.h"
+
+#include "MQTTClient.h"
+
+#define BUF_SIZE                250
+#define MQTT_VERSION            3
+#define COMMAND_TIMEOUT_MS      1000
+#define IS_CLEAN_SESSION        1
+#define NOT_RETAINED_MSG        0
+
+static MQTTClient client;
+static Network network;
+static bool is_initialized = false;
+static unsigned char buf[BUF_SIZE];
+static unsigned char readbuf[BUF_SIZE];
+
+
+static unsigned get_qos(const char *str)
+{
+    int qos = atoi(str);
+
+    switch (qos) {
+        case 1:     return QOS1;
+        case 2:     return QOS2;
+        default:    return QOS0;
+    }
+}
+
+static void _on_msg_received(MessageData *data)
+{
+    printf("mqtt_paho_example: message received on topic"
+           " %.*s: %.*s\n\n",
+           (int)data->topicName->lenstring.len,
+           data->topicName->lenstring.data, (int)data->message->payloadlen,
+           (char *)data->message->payload);
+}
+
+static int _cmd_con(int argc, char **argv)
+{
+    if (argc < 7) {
+        printf(
+            "usage: %s <ipv6 addr> <port> <clientid> <user> <password> <KeepAliveInterval in sec>\n",
+            argv[0]);
+        return 1;
+    }
+
+    char *remote_ip = argv[1];
+    int port = atoi(argv[2]);
+
+    /* ensure client isn't connected in case of a new connection */
+    if (client.isconnected) {
+       printf("client already connected, disconnecting it\n");
+       MQTTDisconnect(&client);
+    }
+
+    if (!is_initialized) {
+        MQTTNetworkInit(&network);
+        printf("Launching MQTT Task\n");
+        MQTTStartTask(&client);
+        is_initialized = true;
+    }
+
+    printf("Trying to connect to %s , port: %d\n",
+           remote_ip, port);
+    if (MQTTNetworkConnect(&network, remote_ip, port)) {
+        printf("error: Unable to connect to %s:%d\n", remote_ip,
+               port);
+        return 1;
+    }
+
+    MQTTClientInit(&client, &network, COMMAND_TIMEOUT_MS, buf, BUF_SIZE, readbuf,
+                   BUF_SIZE);
+
+    MQTTPacket_connectData data = MQTTPacket_connectData_initializer;
+    data.MQTTVersion = MQTT_VERSION;
+    data.clientID.cstring = argv[3];
+    data.username.cstring = argv[4];
+    data.password.cstring = argv[5];
+    data.keepAliveInterval = atoi(argv[6]);
+    data.cleansession = IS_CLEAN_SESSION;
+    printf("Connecting to %s %d\n", remote_ip, port);
+
+    int ret = MQTTConnect(&client, &data);
+    if (ret) {
+        printf("error: Unable to connect client %d\n", ret);
+    }
+    else {
+        printf("Connection successfully\n");
+    }
+    return ret;
+}
+
+static int _cmd_discon(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    int res = MQTTDisconnect(&client);
+    if (res) {
```

Comment:
```suggestion
    if (res < 0) {
```

Reply:
MQTTDisconnect returns a negative integer in case of an error. So this should be changed.